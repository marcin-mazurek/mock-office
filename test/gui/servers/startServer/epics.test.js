/* eslint-disable global-require */
describe('startServerEpic', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test('actions snapshot when response object hasn\'t got errors', (done) => {
    const configureMockStore = require('redux-mock-store').default;
    const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
    const initStart = require('../../../../src/gui/servers/startServer/actions').default;
    jest.mock('../../../../src/gui/api', () => ({
      startServer() {
        return Promise.resolve({
          data: { id: 'some id' }
        });
      }
    }));
    // eslint-disable-next-line global-require
    const startServerEpic = require('../../../../src/gui/servers/startServer/epics').default;

    const epicMiddleware = createEpicMiddleware(startServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore();
    store.dispatch(initStart('some id'));

    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  it('actions snapshot when on api request errors', (done) => {
    const configureMockStore = require('redux-mock-store').default;
    const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
    const initStart = require('../../../../src/gui/servers/startServer/actions').default;
    jest.mock('../../../../src/gui/api', () => ({
      startServer() {
        return Promise.resolve({ error: 'error message' });
      }
    }));
    const startServerEpic = require('../../../../src/gui/servers/startServer/epics').default;

    const epicMiddleware = createEpicMiddleware(startServerEpic);
    const mockStore = configureMockStore([epicMiddleware]);
    const store = mockStore();
    store.dispatch(initStart('some id'));

    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
