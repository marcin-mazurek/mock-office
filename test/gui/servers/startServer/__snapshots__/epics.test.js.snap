// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`startServerEpic actions snapshot when on api request errors 1`] = `
Array [
  Object {
    "id": "some id",
    "type": "startServer/INITIALIZED",
  },
  Object {
    "notification": Object {
      "text": "error message",
      "type": "error",
    },
    "type": "notifications/ADD",
  },
]
`;

exports[`startServerEpic actions snapshot when response object hasn't got errors 1`] = `
Array [
  Object {
    "id": "some id",
    "type": "startServer/INITIALIZED",
  },
  Object {
    "id": "some id",
    "type": "entities/START_SERVER",
  },
]
`;
